src/components/Router/Router.tsx

```diff
@@ -1,10 +1,55 @@
-import React from 'react';
-import { BrowserRouter, HashRouter } from 'react-router-dom';
-import type { BrowserRouterProps, HashRouterProps } from 'react-router-dom';
-
-export default function Router(props: BrowserRouterProps | HashRouterProps) {
-  if (import.meta.env.APP_PUBLIC_GITHUB_PAGES) {
-    return <HashRouter {...props} />;
-  }
-  return <BrowserRouter {...props} />;
+import React, { useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { createBrowserRouter, createHashRouter, createRoutesFromElements, Route, RouterProvider } from 'react-router-dom';
+
+import ErrorPage from '../ErrorPage/ErrorPage';
+import Root from '../Root/Root';
+
+import Layout from '#src/containers/Layout/Layout';
+import About from '#src/screens/About/About';
+import Home from '#src/screens/Home/Home';
+import Movie from '#src/screens/Movie/Movie';
+import Playlist from '#src/screens/Playlist/Playlist';
+import Search from '#src/screens/Search/Search';
+import Series from '#src/screens/Series/Series';
+import User from '#src/screens/User/User';
+
+type Props = {
+  error?: Error | null;
+};
+
+export default function Router({ error }: Props) {
+  const { t } = useTranslation();
+
+  const router = useMemo(() => {
```

Comment:
Maybe add a `@todo` that explains why the routes are currently defined in a memo. Ideally, they would be defined outside the component definition. 

We should also consider moving the Router component to the containers folder or placing it directly in the src directory.

Reply:
I have removed the useMemo, it was unnecessary because the router only renders when there is an error which makes the useMemo useless :P